apiserver/pkg/manager/resource_manager.go

```diff
@@ -181,6 +189,101 @@ func (r *ResourceManager) DeleteCluster(ctx context.Context, clusterName string,
 	return nil
 }
 
+func (r *ResourceManager) CreateJob(ctx context.Context, apiJob *api.RayJob, jobNamespace string) (*v1alpha1.RayJob, error) {
+	computeTemplateMap := make(map[string]*api.ComputeTemplate)
+	var err error
+
+	if apiJob.ClusterSpec != nil && len(apiJob.ClusterSelector) == 0 {
+		// populate cluster map
+		computeTemplateMap, err = r.populateComputeTemplate(ctx, apiJob.ClusterSpec, jobNamespace)
+		if err != nil {
+			return nil, util.NewInternalServerError(err, "Failed to populate compute template for (%s/%s)", jobNamespace, apiJob.JobId)
+		}
+	}
+
+	// convert *api.Cluster to v1alpha1.RayCluster
+	rayJob := util.NewRayJob(apiJob, computeTemplateMap)
```

Comment:
can not we build directly from `apiJob`? any reason to populate `computeTemplateMap` separately?

Reply:
computeTemplateMap is used to construct clusterSpec

Reply:
this is minor and ok.